### PR TITLE
Part18 all tasks are done

### PR DESCRIPTION
**Pull request by alexkulesh**

### CODE REVIEW

File: FirstSolution/Part18Task0/Task.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            foreach (var i in dictionary)
+            {
+                Console.WriteLine(i);
+            }
+
+            string? text = Convert.ToString(Console.ReadLine());
+            string?[] newText = text.Split(" ");
+            var query = newText.Distinct();
+
+            foreach (var i in query)
```
Comment (Alex19777) [inline FirstSolution/Part18Task0/Task.cs]: Rename i variable here and in all same cases

File: FirstSolution/Part18Task1/Task.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            foreach (var i in products)
+            {
+                i.printInfo();
+            }
+
+            foreach (var i in products)
+            {
+                i.Price += 100;
+                i.printInfo();
+            }
```
Comment (Alex19777) [inline FirstSolution/Part18Task1/Task.cs]: You need to change one price

File: FirstSolution/Part18Task1/Task.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+            products.RemoveAt(products.Count - 1);
+
+            var printProducts = from product in products
+                select new {product};
```
Comment (Alex19777) [inline FirstSolution/Part18Task1/Task.cs]: Try to use methods with lambda expression

File: FirstSolution/Part18Task1/Task.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+            for (int i = 0; i < arrayOfProducts1.Length; i++)
+            {
+                arrayOfProducts1[i] = products[i];
+                Console.WriteLine($"Name - {arrayOfProducts1[i].Name}; Price - {arrayOfProducts1[i].Price};");
+            }
+
+            Console.WriteLine("Array 2");
+            var arrayOfProducts2 = (from product in products
+                select product).ToArray();
```
Comment (Alex19777) [inline FirstSolution/Part18Task1/Task.cs:72]: Same here

File: FirstSolution/Part18Task1/Task.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+using Part15Task3;
+
+namespace Part18Task1
+{
+    
+    // а) Создать динамический массив, содержащий объекты класса Product 
+    // б) Распечатать его содержимое используя for each. 
+    // в) Изменить цену одного продукта на 100. 
+    // г) Удалить последний продукт. 
+    // д) Распечатать его содержимое используя LINQ.
```
Comment (Alex19777) [inline FirstSolution/Part18Task1/Task.cs:13]: Where is this implementation?

File: FirstSolution/Part18Task4/Task.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+            Dictionary<string, double> dictionaryOfProducts = new()
+            {
+                {products[0].Name, products[0].Price},
+                {products[1].Name, products[1].Price},
+                {products[2].Name, products[2].Price},
+                {products[3].Name, products[3].Price},
+                {products[4].Name, products[4].Price},
+                {products[5].Name, products[5].Price}
+            };
```
Comment (Alex19777) [inline FirstSolution/Part18Task4/Task.cs]: Leave just Dictionary, remove list

File: FirstSolution/Part18Task4/Task.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            PrintInfo1(dictionaryOfProducts);
+            PrintInfo2(dictionaryOfProducts);
+            PrintInfo3(dictionaryOfProducts);
+        }
+
+        private static void PrintInfo1(Dictionary<string, double> dictionaryOfProducts)
+        {
+            foreach (var i in dictionaryOfProducts)
+            {
+                Console.WriteLine(i);
```
Comment (Alex19777) [inline FirstSolution/Part18Task4/Task.cs]: Do this writeLine write Key and Value of Dictionary?

File: FirstSolution/Part18Task4/Task.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            foreach (var i in dictionaryOfProducts)
+            {
+                Console.WriteLine($"{i.Key}");
+            }
+        }
+
+        private static void PrintInfo3(Dictionary<string, double> dictionaryOfProducts)
+        {
+            var selectedProducts = from product in dictionaryOfProducts
+                where product.Value > 300
```
Comment (Alex19777) [inline FirstSolution/Part18Task4/Task.cs:59]: Do you need value more than 300 here?

File: FirstSolution/Part18Task5/Task.cs [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+            Dictionary<string, double> dictionaryOfProducts = new()
+            {
+                {products[0].Name, products[0].Price},
+                {products[1].Name, products[1].Price},
+                {products[2].Name, products[2].Price},
+                {products[3].Name, products[3].Price},
+                {products[4].Name, products[4].Price},
+                {products[5].Name, products[5].Price}
+            };
```
Comment (Alex19777) [inline FirstSolution/Part18Task5/Task.cs:33]: Leave one dictionary

File: FirstSolution/Part18Task4/Task.cs (hunk 8 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task4/Task.cs:59]: I just need some value

File: FirstSolution/Part18Task1/Task.cs (hunk 2 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task1/Task.cs]: Done

File: FirstSolution/Part18Task1/Task.cs (hunk 3 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task1/Task.cs]: Done

File: FirstSolution/Part18Task1/Task.cs (hunk 4 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task1/Task.cs:72]: It is necesary to make it that way

File: FirstSolution/Part18Task1/Task.cs (hunk 5 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task1/Task.cs:13]: Done

File: FirstSolution/Part18Task4/Task.cs (hunk 6 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task4/Task.cs]: Done

File: FirstSolution/Part18Task0/Task.cs (hunk 1 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task0/Task.cs]: Done

File: FirstSolution/Part18Task4/Task.cs (hunk 7 above)
Comment (alexkulesh) [reply FirstSolution/Part18Task4/Task.cs]: Yes it does

**Review by Alex19777 — APPROVED** [review]

**Pull request merged**